lib/Sema/CSGen.cpp

```diff
@@ -1310,18 +1310,16 @@ namespace {
         return nullptr;
       }
 
-      // Compute the gradient type.
-      auto originalParams = originalTy->getParams();
-      auto *genSig = originalTy->getOptGenericSignature();
-      // Collect differentiation parameter types.
-      SmallVector<TupleTypeElt, 8> diffParamTypes;
+      // Collect differentiation parameter indices.
+      AutoDiffParameterIndices *checkedWrtParamIndices =
```

Comment:
```suggestion
      auto *checkedWrtParamIndices =
```
Nit: redundant type information.